frontend/src/api/client.ts

```diff
@@ -12,23 +12,29 @@
     See the License for the specific language governing permissions and
     limitations under the License.
 */
+interface ClientConfig extends RequestInit {
+  body?: any;
+  stringifyBody?: boolean;
+  parseResponseBodyAs?: "json" | "text" | "blob";
+  omitContentType?: boolean;
+}
 
 export const client = async (
-  endpoint,
-  { body, method, headers, stringifyBody=true, parseResponseBodyAs="json", omitContentType=false, ...customConfig } = {}
+  endpoint: string,
+ { body, method, headers, stringifyBody = true, parseResponseBodyAs = "json", omitContentType = false, ...customConfig }: ClientConfig = {}
 ) => {
-  let customHeaders = {...headers};
-  
+  const customHeaders: { [k: string]: any  } = { ...headers };
```

Comment:
In order to avoid using `any` here. Change this line to:

```ts
const customHeaders = { ...headers } as Record<string,string>;
```

We are casting {..headers} to `Record<string,string>` because it is one of the types of HeaderInit and the one we actually use. It's actually very similar to what you wrote. 

Also, could you add a comment like 

```ts
// { ...headers } is used here in case headers is undefined, which will convert it to an empty object { }
```

Reply:
Done!